utils/crypto/bls/blstest/bls_test.go

```diff
@@ -0,0 +1,555 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bls_test
+
+import (
+	"io"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signers/localsigner"
+
+	jsonrpc "github.com/ava-labs/avalanchego/utils/crypto/bls/signers/json-rpc"
+)
+
+type Signer interface {
+	bls.Signer
+	io.Closer
+}
+
+type jsonRPCSigner struct {
+	*jsonrpc.Client
+	server *jsonrpc.Server
+}
+
+func (s jsonRPCSigner) Close() error {
+	return s.server.Close()
+}
+
+type localSigner struct {
+	*localsigner.LocalSigner
+}
+
+func (localSigner) Close() error {
+	return nil
+}
+
+var localSignerFn = func() (Signer, error) {
+	signer, err := localsigner.NewSigner()
+	return &localSigner{LocalSigner: signer}, err
+}
+
+var serverSignerFn = func() (Signer, error) {
+	// do I need to make sure the server gets closed properly?
+	service := jsonrpc.NewSignerService()
+	server, err := jsonrpc.Serve(service)
+	if err != nil {
+		return nil, err
+	}
+
+	url := url.URL{
+		Scheme: "http",
+		Host:   server.Addr().String(),
+	}
+
+	client := jsonrpc.NewClient(url)
+
+	return jsonRPCSigner{server: server, Client: client}, nil
+}
+
+var signerFns = []func() (Signer, error){
+	localSignerFn,
+	serverSignerFn,
+}
+
```

Comment:
I think the tests in this file are a bit complex and also are now testing integration across the `jsonrpc` signer and `localsigner` which I feel like is unnecessary - if they correctly implement the `Signer` interface they're just returning plain data types and not interfaces so the operations in `bls` should still work across them. I think if we go with the earlier comment's suggestion we can probably get rid of a lot of these tests.